src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx

```diff
@@ -1,5 +1,181 @@
+
+const ByParticipant = () => {
+  const { t } = useTranslation('app');
+  const { appletId, subjectId: respondentSubjectId } = useParams();
+  const { useSubject, useSubjectStatus } = users;
+  const isLoadingRespondentSubject = useSubjectStatus() !== 'success';
+  const { result: respondentSubject } = useSubject() ?? {};
+  const [expandedViewsData, setExpandedViewsData] = useState<
+    Record<string, TargetSubjectsByRespondent | undefined>
+  >({});
+  const [expandedViewsLoading, setExpandedViewsLoading] = useState<Record<string, boolean>>({});
+
+  const {
+    execute: fetchActivities,
+    isLoading: isLoadingActivities,
+    value: fetchedActivities,
+  } = useAsync(getAppletRespondentSubjectActivitiesApi, { retainValue: true });
+
+  const activities = fetchedActivities?.data.result ?? [];
+
+  const handleRefetchExpandedView = useCallback(
+    async (activityOrFlowId: string) => {
+      if (!respondentSubjectId) return;
+
+      setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: true }));
+
+      try {
+        const { data } = await getTargetSubjectsByRespondentApi({
+          activityOrFlowId,
+          subjectId: respondentSubjectId,
+        });
+        // Add or update the correspondent element in expandedViewsData
+        setExpandedViewsData((prev) => ({ ...prev, [activityOrFlowId]: data.result }));
+      } finally {
+        setExpandedViewsLoading((prev) => ({ ...prev, [activityOrFlowId]: false }));
+      }
+    },
+    [respondentSubjectId],
+  );
+
+  const handleRefetchActivities = useCallback(() => {
+    // Avoid fetching activities for respondent if respondent is a limited account
+    if (!appletId || !respondentSubject?.id || !respondentSubject.userId) return;
+
+    fetchActivities({ appletId, subjectId: respondentSubject.id });
+  }, [appletId, fetchActivities, respondentSubject]);
+
+  const handleRefetchAll = useCallback(() => {
+    handleRefetchActivities();
+
+    // Refresh target subject data for any expanded views
+    Object.entries(expandedViewsData).forEach(([id, viewData]) => {
+      if (viewData) handleRefetchExpandedView(id);
+    });
+  }, [handleRefetchActivities, expandedViewsData, handleRefetchExpandedView]);
+
+  const {
+    getActionsMenu,
+    onClickAssign,
+    onClickNavigateToData,
+    isLoading: isLoadingHook,
+    modals,
+  } = useAssignmentsTab({
+    appletId,
+    respondentSubject,
+    handleRefetch: handleRefetchAll,
+    dataTestId,
+  });
+
+  const handleClickNavigateToData = (
+    activityOrFlow: ParticipantActivityOrFlow,
+    targetSubject: RespondentDetails,
+  ) => {
+    if (!respondentSubject) return;
+
+    onClickNavigateToData(activityOrFlow, targetSubject.id);
+  };
+
+  const handleClickToggleExpandedView = useCallback(
+    (isExpanded: boolean, activityOrFlowId: string) => {
+      if (!respondentSubject) return;
+
+      if (isExpanded) {
+        handleRefetchExpandedView(activityOrFlowId);
```

Comment:
Adding `void` before the async function call indicates that we are intentionally ignoring the result/error
```suggestion
        void handleRefetchExpandedView(activityOrFlowId);
```

Reply:
I also think we should differentiate between refetching after the assign drawer closes, and before expanding the card. More often than not, the entries in this list will be the same after expanding the card but since we don't cache the list, it blocks the card from expanding.

Considering that, and the fact that this can be an expensive call depending on the number of results, I think we should cache the results of the call to `getTargetSubjectsByRespondentApi` for some time so that it doesn't perform a fetch on every expand. The cache can be ignored when fetching after the assign drawer closes

Reply:
Since we don't have a pre-existing pattern for caching of endpoints in the Admin App (that I have been able to find, at least), rather than attempt to pull off a new caching system helter-skelter, I spoke to @mbanting and we agreed to split this performance optimization into a post-MI Assign launch fast-follow.

Reply:
Created [M2-8051](https://mindlogger.atlassian.net/browse/M2-8051).